setup/steps/create_log_explain_helper.go

```diff
@@ -0,0 +1,151 @@
+package steps
+
+import (
+	"errors"
+	"fmt"
+
+	survey "github.com/AlecAivazis/survey/v2"
+	"github.com/lib/pq"
+	s "github.com/pganalyze/collector/setup/state"
+	"github.com/pganalyze/collector/setup/util"
+)
+
+var CreateLogExplainHelper = &s.Step{
+	Kind:        s.AutomatedExplainStep,
+	Description: "Create log-based EXPLAIN helper function",
+	Check: func(state *s.SetupState) (bool, error) {
+		logExplain, err := util.UsingLogExplain(state.CurrentSection)
+		if err != nil {
+			return false, err
+		}
+		if !logExplain {
+			return true, nil
+		}
+		monitoredDBs, err := getMonitoredDBs(state)
+		if err != nil {
+			return false, err
+		}
+
+		for _, db := range monitoredDBs {
+			dbRunner := state.QueryRunner.InDB(db)
+			isValid, err := util.ValidateHelperFunction("explain", dbRunner)
+			if !isValid || err != nil {
+				return isValid, err
+			}
+		}
+		return true, nil
+	},
+	Run: func(state *s.SetupState) error {
+		var doCreate bool
+		if state.Inputs.Scripted {
+			if !state.Inputs.CreateExplainHelper.Valid || !state.Inputs.CreateHelperFunctions.Bool {
+				return errors.New("create_explain_helper flag not set and helper function does not exist or does not match expected signature")
+			}
+			doCreate = state.Inputs.CreateHelperFunctions.Bool
+		} else {
+			err := survey.AskOne(&survey.Confirm{
+				Message: "Create (or update) EXPLAIN helper function in each monitored database (will be saved to Postgres)?",
+				Default: false,
+			}, &doCreate)
+			if err != nil {
+				return err
+			}
+		}
+
+		if !doCreate {
+			return nil
+		}
+		monitoredDBs, err := getMonitoredDBs(state)
+		if err != nil {
+			return err
+		}
+		for _, db := range monitoredDBs {
+			err := createHelperInDB(state, db)
+			if err != nil {
+				return err
+			}
+		}
+		return nil
+	},
+}
+
+func createHelperInDB(state *s.SetupState, db string) error {
+	dbRunner := state.QueryRunner.InDB(db)
+	isValid, err := util.ValidateHelperFunction("explain", dbRunner)
+	if err != nil {
+		return err
+	}
+	if isValid {
+		return nil
+	}
+	userKey, err := state.CurrentSection.GetKey("db_username")
+	if err != nil {
+		return err
+	}
+	pgaUser := userKey.String()
+
+	return dbRunner.Exec(
+		fmt.Sprintf(
+			`CREATE SCHEMA IF NOT EXISTS pganalyze; GRANT USAGE ON SCHEMA pganalyze TO %s;`,
+			pq.QuoteIdentifier(pgaUser),
+		) + `
+CREATE OR REPLACE FUNCTION pganalyze.explain(query text, params text[]) RETURNS text AS
```

Comment:
Might be slightly better if we move this to the same location as the helper checksums (`util.ValidateHelperFunction`)